.azure-devops/test/jobs/steps/parallel.yml

```diff
@@ -0,0 +1,66 @@
+steps:
+    - script: |
+        brew update
+      displayName: 'Update brew'
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+
+      # Install macOS required dependencies
+    - script: |
+        set -e
+        brew install xmlstarlet ant ant-contrib tree gnu-sed
+      displayName: "macOS: install dependencies"
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+
+    - script: |
+        echo "##vso[task.setvariable variable=OPENJDK_BINARY]$PIPELINE_WORKSPACE/openjdkbinary";
+        echo "##vso[task.setvariable variable=TEST_JDK_HOME]$PIPELINE_WORKSPACE/openjdkbinary/j2sdk-image";
+        echo "##vso[task.setvariable variable=TEST_DIR]$COMMON_TESTRESULTSDIRECTORY/openjdk-tests"
+        echo "##vso[task.setvariable variable=TKG_DIR]$COMMON_TESTRESULTSDIRECTORY/openjdk-tests/TKG"
+      displayName: "Set environment varaibles"
+    - script: |
+        echo "TEST_JDK_HOME ${TEST_JDK_HOME}";
+        echo "TEST_DIR ${TEST_DIR}";
+        echo "TEST_DIR ${TEST_DIR}";
+        echo "OPENJDK_BINARY ${OPENJDK_BINARY}";
+        echo "ADOPT_TEST_REPO_BRANCH ${ADOPT_TEST_REPO_BRANCH}";
+        echo "ADOPT_TEST_REPO_URI ${ADOPT_TEST_REPO_URI}";
+      displayName: "Print environment varaibles"
+
+    # clone openjdk-test repo
+    - script: |
+        cd $COMMON_TESTRESULTSDIRECTORY
+        git clone -b master --depth 1 https://github.com/AdoptOpenJDK/openjdk-tests
+      displayName: "Clone openjdk test repo"
+
+    - task: DownloadPipelineArtifact@2
```

Comment:
do we still need this step if you are going to run ` ./get.sh -s $PIPELINE_WORKSPACE -t $TEST_DIR -p x86-64_mac -r upstream`?

Reply:
yes, we do. This `DownloadPipelineArtifact ` downloads archived SDK, whereas  `./get.sh -s $PIPELINE_WORKSPACE -t $TEST_DIR -p x86-64_mac -r upstream` downloads test framework (TKG) and other test materials. It does not download SDK in this case.